stash_api/app/controllers/stash_api/datasets_controller.rb

```diff
@@ -41,13 +41,25 @@ def create
       end
     end
 
+    # rubocop:disable Metrics/MethodLength
     # get /datasets
     def index
-      datasets = paged_datasets
+      query_hash = {}
+      query_hash['stash_engine_curation_activities.status'] = params['curationStatus'] if params.key?('curationStatus')
+      if params.key?('publicationISSN')
+        query_hash['stash_engine_internal_data.data_type'] = 'publicationISSN'
+        query_hash['stash_engine_internal_data.value'] = params['publicationISSN']
+      end
+      @datasets = StashEngine::Identifier.select('stash_engine_curation_activities.*, stash_engine_internal_data.*, stash_engine_identifiers.*')
+        .joins(:curation_activities, :internal_data)
```

Comment:
It turns out that these JOINs cause problems because they only select identifiers that also have curation activity and internal data.  Maybe we won't have some (at least internal data) for existing records from CDL.  Not sure, but this will let through all identifiers.

```.joins('LEFT JOIN stash_engine_curation_activities ON stash_engine_identifiers.id = stash_engine_curation_activities.identifier_id')
        .joins('LEFT JOIN stash_engine_internal_data ON stash_engine_identifiers.id = stash_engine_internal_data.identifier_id')
        .where(query_hash)```

Reply:
I was just really hoping to be able to use Rails-y syntax and not have to manually write out the queries, but perhaps LEFT JOIN is the only way. I'll fix this.